vnext/Shared/Networking/WinRTHttpResource.cpp

```diff
@@ -55,6 +58,138 @@ WinRTHttpResource::WinRTHttpResource(IHttpClient &&client) noexcept : m_client{s
 
 WinRTHttpResource::WinRTHttpResource() noexcept : WinRTHttpResource(winrt::Windows::Web::Http::HttpClient{}) {}
 
+#pragma region IWinRTHttpRequestFactory
+
+IAsyncOperation<HttpRequestMessage> WinRTHttpResource::CreateRequest(
+    HttpMethod &&method,
+    Uri &&uri,
+    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, IInspectable> props) noexcept /*override*/ {
+  auto request = HttpRequestMessage{std::move(method), std::move(uri)};
+  for (auto prop : props) {
+    request.Properties().Insert(prop.Key(), prop.Value());
+  }
```

Comment:
Are there any guarantees that prop.Key() is unique in the incomming props. And subsequently what would happen if duplicate keys are added to the Properties IMap<> WIll it fail or last write wins?

Reply:
IMap provides such guarantees.

Last write wins.